# Project 01: Exploring and Visualizing Iris Dataset

## Introduction

Data analysis is a fundamental step in the field of Data Science. Before applying advanced analytical techniques or machine learning algorithms, it is important to understand the structure, quality, and characteristics of the data. Exploratory Data Analysis (EDA) helps analysts identify patterns, trends, relationships, and potential issues within a dataset.

The Iris Dataset is one of the most widely used datasets in data science and machine learning education. It contains measurements of iris flowers collected from three different species: Iris Setosa, Iris Versicolor, and Iris Virginica. Due to its simplicity and well-structured nature, the dataset is commonly used to learn data exploration and visualization techniques.

This project focuses on understanding the dataset through data inspection, summarization, and visualization. By using Python libraries such as Pandas, Matplotlib, and Seaborn, we will analyze the dataset and create visual representations that help reveal important patterns and relationships among the variables.

---

## Problem Statement

Understanding a dataset is an essential step before performing any analytical or predictive task. Without proper exploration, it becomes difficult to identify the distribution of variables, relationships between features, and potential anomalies within the data.

The problem addressed in this project is to explore and visualize the Iris Dataset in order to gain a better understanding of its structure and characteristics. Different visualization techniques will be used to examine feature distributions, identify patterns among flower species, and observe the spread of numerical variables.

---

## Project Objectives

The main objectives of this project are:

- To load and inspect the Iris Dataset using the Pandas library.
- To understand the structure and dimensions of the dataset.
- To examine the available features and their data types.
- To perform basic data summarization and inspection.
- To create visualizations using Matplotlib and Seaborn.
- To analyze relationships between variables through scatter plots.
- To examine feature distributions using histograms.
- To identify variability and potential outliers using box plots.
- To derive meaningful insights from the dataset through visual analysis.

---

## Tools and Libraries Used

The following tools and libraries are used in this project:

- Python
- Pandas
- Matplotlib
- Seaborn
- Jupyter Notebook

These libraries provide powerful capabilities for data manipulation, analysis, and visualization, making them essential tools in the data science workflow.

# Dataset Understanding and Description

## About the Dataset

The Iris Dataset is a multivariate dataset introduced by the British statistician and biologist Ronald Fisher in 1936. It is one of the most frequently used datasets for demonstrating statistical analysis, data visualization, and machine learning concepts.

The dataset contains measurements collected from iris flowers belonging to three different species:

1. Iris Setosa
2. Iris Versicolor
3. Iris Virginica

Each observation in the dataset represents a single flower sample and includes measurements of its sepal and petal dimensions.

---

## Dataset Features

The dataset contains six columns:

### 1. Id
A unique identifier assigned to each flower sample.

### 2. SepalLengthCm
The length of the sepal measured in centimeters.

### 3. SepalWidthCm
The width of the sepal measured in centimeters.

### 4. PetalLengthCm
The length of the petal measured in centimeters.

### 5. PetalWidthCm
The width of the petal measured in centimeters.

### 6. Species
The species classification of the iris flower:
- Iris-setosa
- Iris-versicolor
- Iris-virginica

---

## Importance of the Dataset

The Iris Dataset is widely used because it is simple, clean, and easy to understand. It provides an excellent opportunity to learn how to:

- Explore a dataset.
- Perform descriptive analysis.
- Create meaningful visualizations.
- Understand relationships between variables.
- Develop a foundation for future machine learning tasks.

The dataset serves as an ideal starting point for beginners entering the field of Data Science and Analytics.

## Import Required Libraries

In [7]:
# Import libraries for data analysis and visualization

import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

## Load Dataset

In this section, the Iris dataset is loaded into a Pandas DataFrame. Loading the dataset is the first step in any data analysis project because it allows us to inspect, manipulate, and analyze the data using Python.

The dataset used in this project contains measurements of iris flowers from three different species. After loading the dataset, we will examine its structure and basic characteristics.

In [11]:
# Load the Iris dataset
df = pd.read_csv("../data/raw/iris.csv")

# Display the first five rows of the dataset
df.head()

,Id,SepalLengthCm,SepalWidthCm,PetalLengthCm,PetalWidthCm,Species
0,1,5.1,3.5,1.4,0.2,Iris-setosa
1,2,4.9,3.0,1.4,0.2,Iris-setosa
2,3,4.7,3.2,1.3,0.2,Iris-setosa
3,4,4.6,3.1,1.5,0.2,Iris-setosa
4,5,5.0,3.6,1.4,0.2,Iris-setosa


### Observation

The dataset was loaded successfully. The preview shows that the dataset contains six columns, including an identifier column (Id), four numerical measurement features, and one categorical feature (Species).

The Species column contains different iris flower categories, while the remaining numerical columns represent sepal and petal measurements in centimeters.

## Dataset Structure

Understanding the structure of a dataset is an important step before performing any analysis. It helps identify the number of observations, available features, and the overall organization of the data.

The following analysis examines:

- Dataset dimensions
- Column names
- Sample records

In [10]:
# Display dataset dimensions
print("Dataset Shape:")
print(df.shape)

Dataset Shape:
(150, 6)


### Observation

The dataset contains 150 rows and 6 columns. Each row represents an individual iris flower sample, while the columns represent flower measurements and species information.

In [17]:
# Display column names
print("Column Names:", df.columns.tolist())

Column Names: ['Id', 'SepalLengthCm', 'SepalWidthCm', 'PetalLengthCm', 'PetalWidthCm', 'Species']


### Observation

The dataset consists of six columns: Id, SepalLengthCm, SepalWidthCm, PetalLengthCm, PetalWidthCm, and Species.

The Id column serves as a unique identifier, while the remaining columns contain flower measurements and species classification information.